src/preferences.tsx

```diff
@@ -17,50 +17,50 @@ export interface IPreferences {
   routerType: "browser" | "memory";
 }
 
-const { condaStoreConfig = {} } =
-  typeof window !== "undefined" && (window as any);
+let condaStoreConfig: any = {};
+if (typeof window !== "undefined" && "condaStoreConfig" in window) {
+  condaStoreConfig = window.condaStoreConfig;
+}
 
 export const prefDefault: Readonly<IPreferences> = {
   apiUrl:
-    process.env.REACT_APP_API_URL ??
     condaStoreConfig.REACT_APP_API_URL ??
```

Comment:
The Conda Store docs say that [condaStoreConfig can be used to configure the UI app at runtime](https://conda.store/conda-store-ui/how-tos/configure-ui). 

I take this to also imply that run-time should override build-time configuration. 

If not, it is impossible to set a config variable in the way that I do in the `memory-router-test.html` file in this PR if that variable was already defined as an environment variable at build time.

So for example, if in my .env file I have:

```
REACT_APP_ROUTER_TYPE=browser
```

And then in the my app index html page I have:

```
<script>
window.condaStoreConfig = {
  REACT_APP_ROUTER_TYPE: "memory"
};
<script>
```

Then the app will actually be loaded with the browser router and not the memory router. I don't think that's what we want, but I may be mistaken.

Reply:
Runtime indeed overwrites/supersedes build time.
This is now how we are passing default config values to conda-store-server vs the hack we used to have before https://github.com/conda-incubator/conda-store/blob/d39e5b47577aa6764b1bf99a0ab675fe8782e969/conda-store-server/conda_store_server/_internal/server/templates/conda-store-ui.html#L11

Reply:
That's interesting. Does that file you sent actually work? Because as far as I can tell, the only way it would work is if the `main.js` file at `static/conda-store-ui/main.js` would have to have been built in an environment in which none of those `REACT_APP_`-prefixed variables were defined at build time. I searched the UI codebase for `condaStoreConfig` to see if I was missing something, but the only place that variable is read is in the `preferences.tsx` file, and in that file, it's pretty clear that the variable from `condaStoreConfig` is only read if the corresponding environment variable were nullish at build-time.

For example, let's look at the following JavaScript:

```js
process.env.REACT_APP_API_URL ??
condaStoreConfig.REACT_APP_API_URL ??
"http://localhost:8080/conda-store/"
```

At build time Webpack will do a string substitution and replace `process.env.REACT_APP_API_URL` with the value in the environment, if it's defined. So, if it's set to "http://localhost:8080/conda-store/", then the above JavaScript is transformed to:

```js
"http://localhost:8080/conda-store/" ??
condaStoreConfig.REACT_APP_API_URL ??
"http://localhost:8080/conda-store/"
```

Since the first argument in the above expression is not "nullish", meaning neither `null` nor `undefined`, then the overall expression reduces to that first argument, i.e.:

```js
"http://localhost:8080/conda-store/"
```

On the other hand, if that environment variable is not defined at build time, the source code is transformed as:

```js
undefined ??
condaStoreConfig.REACT_APP_API_URL ??
"http://localhost:8080/conda-store/"
```

Since the left hand side of the nullish operator is undefined the runtime will try to read the value from `condaStoreConfig`, and only if that too is null or undefined, will it return the last operand.

So the only way the condaStoreConfig variable plays any role at run time is if the corresponding environment variable was not defined at build time.